### ISSUE TITLE
getBoundingClientRect error if element is deleted

### ISSUE THREAD

**Issue by braydonburn** [elrumordelaluz/reactour#239]
Hello,
If the tour element is deleted while it's in the process of opening an error will occur, `Cannot read property 'getBoundingClientRect' of null`
</br>

**Codepen demonstrating the issue:**
Click the `Try it` button to trigger the error
https://codesandbox.io/s/reactour-demo-template-live-u934w

**Comment by Nikki1993** [comment]
Possibly related to the missing componentWillUnmount method wrongfully tagged as UNSAFE. [This PR](https://github.com/elrumordelaluz/reactour/pull/238 ) was just merged to v1, if you don't want to wait for new release you can git clone latest v1 branch and see if it crashes on getBoundingClientRec

**Comment by elrumordelaluz** [comment]
`reactour@1.16.2` published but seems persisting the issue updating the version in your sandbox. Need a little more investigation.

**Comment by Nikki1993** [comment]
@elrumordelaluz are you sure? I just tried the sandboxing link, bumped the version and reloaded the internal browser window and no crash when clicking "Try it"

**Comment by elrumordelaluz** [comment]
cool, so my bad bumping version

**Comment by braydonburn** [comment]
> @elrumordelaluz are you sure? I just tried the sandboxing link, bumped the version and reloaded the internal browser window and no crash when clicking "Try it"

Can you please double check this? I've followed your steps and the error still persists on my end

**Comment by bennnjamin** [comment]
I am having this issue in Microsoft Edge 18.18363 and IE 11. Edge 79+ is working. For now I am disabling the tour if the user is running these browsers as having the website crash or forcing them to download a compatible browser is not a great UX.

**Comment by Nikki1993** [comment]
@braydonburn just did it again, didn't see any error popping up on Chrome 80 but for some reason I can see the error when I go incognito mode. This is weird as I could reproduce the error 100% of the time before in normal browser window. You are correct, the issue is still there.

**Comment by Nikki1993** [comment]
OK, I had a quick look and it seems that this error happens because Tour.js passes `this.helper` but when it's closed really fast it's still null or React resets it to null, not sure exactly. 

We can add rudimentary null checks for this.helper but that's treating symptoms not the problem, however it seems like reactour is in the process of rewrite to hooks on master so perhaps that could be an ideal solution @elrumordelaluz thoughts?

**Comment by Nikki1993** [comment]
Checked the componentDidMount and this.helper is set to
```javascript
this.helper = { current: null }
``` so it seems the ref is not being init properly?

**Comment by Nikki1993** [comment]
It seems like when component dismounts, it calls showStep function and that's when it crashes since this.helper at that point is null. This happens because UNSAFE_componentWillReceiveProps triggers before the componentWillUnmount but the refs have already been cleaned up.

**Comment by elrumordelaluz** [comment]
> however it seems like reactour is in the process of rewrite to hooks

Absolutely right, in fact is yet working with _hooks_ in the `master` branch. It isn't yet published because is still unsolved some _issues_ regarding `actions` in each `step`.

I am thinking in publishing a `beta` to let you try the sandbox.

**Comment by Nikki1993** [comment]
I went ahead and submitted the PR https://github.com/elrumordelaluz/reactour/pull/240 but again this is not ideal as all it's doing is checking if refs are present before executing showStep function. I tested and it seems to work at least for the example case above the error is gone 
and I figured it's fine to do a "hacky" fix as rewrite to hooks is incoming.

**Comment by elrumordelaluz** [comment]
published as 📦 `reactour@1.16.3`. Let me know in any case.
